specs/Print.md

```diff
@@ -0,0 +1,816 @@
+// with the default print settings.
+
+bool AppWindow::PrintToDefaultPrinter()
+{
+  wil::com_ptr<ICoreWebView2_15> webView2_15;
+  CHECK_FAILURE(m_webView->QueryInterface(IID_PPV_ARGS(&webView2_15)));
+
+  wil::com_ptr<ICoreWebView2Environment6> webviewEnvironment6;
+  CHECK_FAILURE(m_webViewEnvironment->QueryInterface(IID_PPV_ARGS(&webviewEnvironment6)));
+
+  wil::com_ptr<ICoreWebView2PrintSettings> printSettings = nullptr;
+  CHECK_FAILURE(webviewEnvironment6->CreatePrintSettings(&printSettings));
+
+  wil::com_ptr<ICoreWebView2PrintSettings2> printSettings2;
+  CHECK_FAILURE(printSettings->QueryInterface(IID_PPV_ARGS(&printSettings2)));
+
+  wil::unique_cotaskmem_string title;
+  CHECK_FAILURE(m_webView->get_DocumentTitle(&title));
+
+  // Prints current web page with the default page and printer settings.
+  CHECK_FAILURE(webView2_15->Print(
+      printSettings2.get(),
+      Callback<ICoreWebView2PrintCompletedHandler>(
+          [title = title.get(), this](HRESULT errorCode, BOOL isSuccessful) -> HRESULT
+          {
+            std::wstring message = L"";
+            if (errorCode == S_OK && isSuccessful)
+            {
+              message = L"Printing " + std::wstring(title) +
+                              L" document to printer is succedded";
+            }
+            else
+            {
+              if (errorCode == S_OK)
+              {
+                message = L"Printer is not available, offline or error state";
+              }
+              else if (errorCode == E_ABORT)
+              {
+                message = L"Printing " + std::wstring(title) +
+                                  L" document already in progress";
+              }
+              else if (errorCode == E_FAIL)
+              {
+                message = L"Printing " + std::wstring(title) +
+                                  L" document to printer is failed";
+              }
+            }
+
+            AsyncMessageBox(message, L"Print to default printer");
+
+            return S_OK;
+        })
+        .Get()));
+  return true;
+}
+```
+
+```c#
+async void PrintToDefaultPrinter ()
+{
+  try
+  {
+    CoreWebView2PrintSettings printSettings = null;
+    printSettings = WebViewEnvironment.CreatePrintSettings();
+
+    string title = webView.CoreWebView2.DocumentTitle;
+
+    // Prints current web page with the default page and printer settings.
+    bool isSuccessful = await webView.CoreWebView2.PrintAsync(printSettings);
+
+    if (isSuccessful)
+    {
+      MessageBox.Show(this, "Printing " + title + " document to printer is succeeded", "Print to default printer");
+    }
+    else
+    {
+      MessageBox.Show(this, "Printer is not available, offline or error state", "Print to default printer");
+    }
+  }
+  catch (NotImplementedException exception)
+  {
+    MessageBox.Show(this, "Printing is failed: " + exception.Message, "Print to default printer");
+  }
```

Comment:
The C# version doesn't catch the `E_ABORT` or `E_FAIL` cases, so if printing is already in progress, the app crashes.

The C# and C++ versions should have equivalent error handling. Otherwise people are going to wonder "Why do you have to check errors in C++ but not C#?" or "How come you can detect printing in progress in C++, but not in C#?" Basically, if the two versions aren't equivalent, people will assume they are different for a reason and then wonder what that reason is.

Reply:
Please see later comment about error handling.